lib/ops/utils.go

```diff
@@ -34,118 +34,165 @@ import (
-	op, _, err = MatchOperation(siteKey, operator, func(op SiteOperation) bool {
-		return op.Type == OperationUpdate && op.IsCompleted()
+	operations, err := operator.GetSiteOperations(siteKey, OperationsFilter{
+		Type:     OperationUpdate,
+		Last:     true,
+		Complete: true,
 	})
 	if err != nil {
-		if trace.IsNotFound(err) {
-			return nil, trace.NotFound("no update operation for %v found", siteKey)
-		}
 		return nil, trace.Wrap(err)
 	}
-	return op, nil
+
+	if len(operations) == 0 {
+		return nil, trace.NotFound("no completed update operation found for %v", siteKey)
+	}
+
+	return (*SiteOperation)(&operations[0]), nil
 }
 
 // GetCompletedInstallOperation returns a completed install operation for the specified site
 func GetCompletedInstallOperation(siteKey SiteKey, operator Operator) (*SiteOperation, error) {
-	op, entry, err := GetInstallOperation(siteKey, operator)
+	operations, err := operator.GetSiteOperations(siteKey, OperationsFilter{
+		Type:     OperationInstall,
+		Last:     true,
+		Complete: true,
+	})
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
-	if entry.IsCompleted() {
-		return op, nil
+	if len(operations) == 0 {
+		return nil, trace.NotFound("no completed install operation found for %v", siteKey)
 	}
-	return nil, trace.NotFound("no completed install operation for %v found", siteKey)
+
+	return (*SiteOperation)(&operations[0]), nil
 }
 
 // GetLastOperation returns the most recent operation and its progress for the specified site
-func GetLastOperation(siteKey SiteKey, operator Operator) (*SiteOperation, *ProgressEntry, error) {
-	operations, err := operator.GetSiteOperations(siteKey)
+func GetLastOperation(siteKey SiteKey, operator Operator) (op *SiteOperation, progress *ProgressEntry, err error) {
+	operations, err := operator.GetSiteOperations(siteKey, OperationsFilter{
+		Last: true,
+	})
 	if err != nil {
 		return nil, nil, trace.Wrap(err)
 	}
+
 	if len(operations) == 0 {
-		return nil, nil, trace.NotFound("no operations found for %v", siteKey)
+		return nil, nil, trace.NotFound("no operation found for %v", siteKey)
 	}
+
 	// backend is guaranteed to return operations in the last-to-first order
-	lastOperation := (*SiteOperation)(&operations[0])
-	progress, err := operator.GetSiteOperationProgress(lastOperation.Key())
+	op = (*SiteOperation)(&operations[0])
+	progress, err = operator.GetSiteOperationProgress(op.Key())
 	if err != nil {
 		return nil, nil, trace.Wrap(err)
 	}
-	return lastOperation, progress, nil
+
+	return op, progress, nil
 }
 
 // GetLastCompletedOperations returns the cluster's last completed operation
-func GetLastCompletedOperation(key SiteKey, operator Operator) (*SiteOperation, *ProgressEntry, error) {
-	operations, err := operator.GetSiteOperations(key)
+func GetLastCompletedOperation(siteKey SiteKey, operator Operator) (op *SiteOperation, progress *ProgressEntry, err error) {
```

Comment:
Nit: I would rename this to `GetLastFinishedOperation` as that's what it's actually searching for.